nixio/pycore/section.py

```diff
@@ -17,43 +20,64 @@ def __init__(self, h5obj):
     @classmethod
     def _create_new(cls, parent, name, type_):
         newentity = super(Section, cls)._create_new(parent, name, type_)
+        newentity._h5obj.create_group("sections")
+        newentity._h5obj.create_group("properties")
         return newentity
 
-    def create_section(self):
+    def create_section(self, name, type_):
+        util.check_entity_name_and_type(name, type_)
+        sections = self._h5obj["sections"]
+        if name in sections:
+            raise exceptions.DuplicateName("create_section")
+        sec = Section._create_new(sections, name, type_)
+        return sec
+
+    @property
+    def parent(self):
```

Comment:
If a Section was created from a group opened via the `metadata` hard link of an `EntityWithMetadata` the parent of that group might not be the group of the parent section but the entity providing the hard link instead.